src/js/tools/drawing/Stroke.js

```diff
@@ -97,8 +97,35 @@
       linePixels = pskl.PixelUtils.getLinePixels(col, this.startCol, row, this.startRow);
     }
 
-    pskl.PixelUtils.resizePixels(linePixels, penSize).forEach(function (point) {
-      targetFrame.setPixel(point[0], point[1], color);
```

Comment:
Below, I implement a (potentially verbose) procedural diamond at each end of the line. It is accomplished by conditionally coloring a square on the basis of the taxicab distance from the center of a square.